src/test/java/org/wildfly/security/auth/client/MatchHostRuleTest.java

```diff
@@ -0,0 +1,120 @@
+package org.wildfly.security.auth.client;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.regex.Pattern;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * @author Tomas Hofman (thofman@redhat.com)
+ */
+public class MatchHostRuleTest {
+
+    @Test
+    public void testDomainName() throws URISyntaxException {
+        Assert.assertTrue(MatchRule.ALL.matchHost("security.wildfly.org").matches(new URI("remote+http://security.wildfly.org:9990")));
+        Assert.assertTrue(MatchRule.ALL.matchHost("security.wildfly.").matches(new URI("remote+http://security.wildfly.org:9990")));
+        Assert.assertTrue(MatchRule.ALL.matchHost("security.wildfly").matches(new URI("remote+http://security.wildfly.org:9990")));
+
+        Assert.assertFalse(MatchRule.ALL.matchHost("wildfly.org").matches(new URI("remote+http://security.wildfly.org:9990")));
+    }
+
+    @Test
+    public void testIPv4() throws URISyntaxException {
+        Assert.assertTrue(MatchRule.ALL.matchHost("127.0.0.1").matches(new URI("remote+http://127.0.0.1:9990")));
+        Assert.assertTrue(MatchRule.ALL.matchHost("127.0.0.").matches(new URI("remote+http://127.0.0.1:9990")));
+        Assert.assertTrue(MatchRule.ALL.matchHost("127.0.0").matches(new URI("remote+http://127.0.0.1:9990")));
```

Comment:
This is not OK because the old RFC for IPv4 explains that the last segment has to cover the number of remaining bits  in the IP address.  So if we're really parsing this value as an IP address then 127.0.0 == 127.0.0.0, which != 127.0.0.1.